resources/git_credentials.rb

```diff
@@ -0,0 +1,35 @@
+resource_name :git_credentials
+
+property :file, String, default: '/root/.git-credentials'
+property :secrets_databag, String, default: node['osl-git']['secrets_databag']
+property :secrets_item, String, default: node['osl-git']['secrets_item']
+
+default_action :create
+
+action :create do
+  with_run_context :root do
+    execute "git config --global credential.helper 'store --file #{new_resource.file}'"
+
+    edit_resource(:template, new_resource.file) do |new_resource|
```

Comment:
Why are you using ``edit_resource`` here?

Reply:
@ramereth The purpose of the `edit_resource` is for all of a Chef run's `git_credentials` resources (that specify the same file) to populate the same template resource; using normal templates (instead of editing only one template resource) would result in them overriding each other, where the last run `git_credentials` resource prevails. The `edit_resource` allows them to merge their values into a single template which will include all of the `git_credentials` resources by the end of the run.

I think it's called an [accumulator pattern](https://blog.dnsimple.com/2017/10/chef-accumulators/).

Reply:
As per our discussion, the accumulator pattern has been removed. The code is preserved (squashed down into the previous commit).